src/Simplify.cpp

```diff
@@ -274,8 +275,16 @@ bool can_prove(Expr e, const Scope<Interval> &bounds) {
     };
     e = RemoveLikelies().mutate(e);
 
+    e = common_subexpression_elimination(e);
+
+    Expr orig = e;
+
     e = simplify(e, true, bounds);
 
+    // When terms cancel, the simplifier doesn't always successfully
+    // kill the dead lets.
+    while (const Let *l = e.as<Let>()) e = l->body;
```

Comment:
Loops without { curly braces } make me nervous, even trivial ones like this. Don't we (generally) prefer all loopy structures to be embraced?